client/src/controllers/BlockController.ts

```diff
@@ -0,0 +1,67 @@
+import { Controller } from '@hotwired/stimulus';
+import { object } from 'prop-types';
+
+/**
+ * Adds the ability to control a block element.
+ *
+ * @example
+ * <div
+ * id="{id}"
+ * data-block
+ * data-controller="w-block"
+ * data-w-block-data-value="{block_json}"
+ * data-w-block-initial-value="{value_json}"
+ * data-w-block-error-value="{error_json}">
+ * </div>
+ */
+
+declare global {
+  interface Window {
+    initBlockWidget?: (id: string) => void; // Declare the initBlockWidget function as optional on the Window interface
+    telepath?: any;
+  }
+}
+
+export class BlockController extends Controller<HTMLElement> {
+  static values = {
+    initial: { type: object, default: {} }, // Initial value for the block
+    data: object, // Unpacked Telepath data
+    error: object, // Block error (optional)
+  };
+
+  declare dataValue: object;
```

Comment:
```suggestion
  /** Packed Telepath data. */
  declare dataValue: object;
```

As per comment above, move the description of values to JSDoc format and above the declaration lines.